src/CommitmentsV2/SFA.DAS.CommitmentsV2.MessageHandlers/EventHandlers/RecordedAct1CompletionPaymentEventHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.Extensions.Logging;
+using NServiceBus;
+using SFA.DAS.CommitmentsV2.Messages.Events;
+using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using SFA.DAS.CommitmentsV2.Types;
+
+namespace SFA.DAS.CommitmentsV2.MessageHandlers.EventHandlers
+{
+    public class RecordedAct1CompletionPaymentEventHandler : IHandleMessages<RecordedAct1CompletionPaymentFakeEvent>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+        private readonly ILogger<RecordedAct1CompletionPaymentEventHandler> _logger;
+
+        public RecordedAct1CompletionPaymentEventHandler(Lazy<ProviderCommitmentsDbContext> dbContext, ILogger<RecordedAct1CompletionPaymentEventHandler> logger)
+        {
+            _dbContext = dbContext;
+            _logger = logger;
+        }
+
+        public async Task Handle(RecordedAct1CompletionPaymentFakeEvent message, IMessageHandlerContext context)
+        {
+            try
+            {
+                if (message.ApprenticeshipId.HasValue)
```

Comment:
It would be good to get logging in the event that the apprenticeshipId has no value, as this is an indicator of something awry in payments.

Reply:
yes meant to do that. slipped my mind